src/CommunityToolkit.Maui/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.ComponentModel;
+
+/// Using root CommunityToolkit.Maui namespace so these extension methods
+/// light up in MauiProgram when CommunityToolkit.Maui namespace is imported 
+/// for UseMauiCommunityToolkit();
+namespace CommunityToolkit.Maui;
+
+/// <summary>
+/// Extension methods for registering Views and ViewModels in <see cref="IServiceCollection"/>
+/// </summary>
+public static class IServiceCollectionExtensions
+{
+	/// <summary>
+	/// Adds a Page or View of the type specified in <typeparamref name="TView"/> and a ViewModel
+	/// of the type specified in <typeparamref name="TViewModel"/> to the specified
+	/// <see cref="IServiceCollection"/> with <see cref="ServiceLifetime.Transient"/> lifetime.
+	/// </summary>
+	/// <typeparam name="TView">The type of the View to add. Constrained to <see cref="BindableObject"/></typeparam>
+	/// <typeparam name="TViewModel">The type of the ViewModel to add. Constrained to 
+	/// <see cref="INotifyPropertyChanged"/></typeparam>
+	/// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+	/// <returns>A reference to this instance after the operation has completed.</returns>
+	/// <remarks>Developers are still responsible for assigning the injected instance of 
+	/// <typeparamref name="TViewModel" />
+	/// to the BindingContext property of <typeparamref name="TView" />.</remarks>
+	public static IServiceCollection AddTransient<TView, TViewModel>(this IServiceCollection services)
+		where TView : BindableObject
+		where TViewModel : class, INotifyPropertyChanged
```

Comment:
Let's remove the `class` from types here and in other places, I'm pretty sure that `INotifyPropertyChanged` is enough
```suggestion
		where TViewModel : INotifyPropertyChanged
```

Reply:
Good catch! Yup, technically a `struct` can be used as a ViewModel.

I've never seen a ViewModel implemented as a `struct` before, but we should certainly still leave that option open since .NET MAUI allows it.

Reply:
It's currently constrained as broadly as possible due to underlying `IServiceCollection` extension methods used:

```csharp
        /// <summary>
        /// Adds a transient service of the type specified in <typeparamref name="TService"/> to the
        /// specified <see cref="IServiceCollection"/>.
        /// </summary>
        /// <typeparam name="TService">The type of the service to add.</typeparam>
        /// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
        /// <returns>A reference to this instance after the operation has completed.</returns>
        /// <seealso cref="ServiceLifetime.Transient"/>
        public static IServiceCollection AddTransient<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TService>(this IServiceCollection services)
            where TService : class
        {
            if (services == null)
            {
                throw new ArgumentNullException(nameof(services));
            }

            return services.AddTransient(typeof(TService));
        }
```

I suppose we could switch to other IServiceCollection methods to register the types, however it's going to add more code to this PR. Additionally, the `[INotifyPropertyChanged]` attribute from CommunityToolkit.Mvvm only supports classes and `ObservableObject` is also a class.

My recommendation is we leave the constraint `where TViewModel : class, INotifyPropertyChanged`. Thoughts?

Reply:
Well if the base implementation restricts to a `class`, we can keep it to match and avoid any runtime exception.

Reply:
Yep, I tried `INotifyPropertyChanged` at first, but because underlying `IServiceCollection` method required reference type, I had to add `class` to the constraint.